rotkehlchen/db/dbhandler.py

```diff
@@ -342,6 +342,19 @@ def connect(self, password: str) -> None:
             self.conn.executescript(script)
         self.conn.execute('PRAGMA foreign_keys=ON')
 
+    def change_password(self, new_password: str) -> None:
+        """Changes the password for the currently logged in user
+        """
+        self.conn.text_factory = str
+        new_password_for_sqlcipher = _protect_password_sqlcipher(new_password)
+        self.conn.executescript(f'PRAGMA rekey="{new_password_for_sqlcipher}";')
+        if self.sqlcipher_version == 3:
+            script = f'PRAGMA rekey="{new_password_for_sqlcipher}"; PRAGMA kdf_iter={KDF_ITER};'
+            try:
+                self.conn.executescript(script)
+            except sqlcipher.OperationalError as e:  # pylint: disable=no-member
+                log.error(f'At change password could not re-key the open database: {str(e)}')
```

Comment:
You will also need to return something from this function. A simple boolean should suffice. So here apart from logging an error you should return `False` and handle it accordingly in the rest api.